analysis_notebooks/Attribute Analysis.ipynb

```diff
@@ -981,6 +984,285 @@
+   "source": [
+    "# att_dis Metric : Distance from object as proxy for interaction\n",
+    "<a id=\"att_dis\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup\n",
+    "<a id=\"att_dis_setup\"></a>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hide_toggle(for_next=True, toggle_text='Show/hide att_dis code')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "categories = dataset.categories\n",
+    "names = dataset.labels_to_names\n",
+    "distances = pickle.load(open(\"results/{}/att_dis.pkl\".format(folder_name), \"rb\"))\n",
+    "topn = 10\n",
+    "\n",
+    "attr_dists = []\n",
+    "attr_distr_dists = []\n",
+    "attr_file_paths = []\n",
+    "attr_indices = []\n",
+    "p_values = []\n",
+    "ratio_idxs = []\n",
+    "xaxis = []\n",
+    "i_list = []\n",
+    "for i in range(len(categories)):\n",
+    "    largest_ratio = -1\n",
+    "    ratio_idx = -1\n",
+    "    for j in range(num_attrs):\n",
+    "        a_dist = np.array([ (chunk[0] / np.sqrt(chunk[1]*chunk[2])) for chunk in distances[i][j] if chunk[2] != 0 and chunk[1] != 0])\n",
+    "        a_file = np.array([ chunk[3] for chunk in distances[i][j] if chunk[2] != 0 and chunk[1] != 0])\n",
+    "        a_index = np.array([ chunk[4] for chunk in distances[i][j] if chunk[2] != 0 and chunk[1] != 0])\n",
+    "        b_dist = []\n",
+    "        b_file = []\n",
+    "        b_index = []\n",
+    "        for k in range(num_attrs):\n",
+    "            b_dist.extend([(chunk[0] / np.sqrt(chunk[1]*chunk[2])) for chunk in distances[i][k] if chunk[2] != 0 and chunk[1] != 0 and j != k])\n",
+    "            b_file.extend([chunk[3] for chunk in distances[i][k] if chunk[2] != 0 and chunk[1] != 0 and j != k])\n",
+    "            b_index.extend([chunk[4] for chunk in distances[i][k] if chunk[2] != 0 and chunk[1] != 0 and j != k])\n",
+    "        \n",
+    "        t, p = stats.ttest_ind(a_dist, b_dist, equal_var=False)\n",
+    "\n",
+    "        a_mean = np.mean(a_dist)\n",
+    "        b_mean = np.mean(b_dist)\n",
+    "        if not np.isnan(a_mean) and not np.isnan(b_mean) and len(a_dist) > 5 and len(b_dist) > 5:\n",
+    "            if (p < 0.05):\n",
+    "                attr_dists.append([a_mean, b_mean])\n",
+    "                attr_distr_dists.append([a_dist, b_dist])\n",
+    "                attr_file_paths.append([a_file, b_file])\n",
+    "                attr_indices.append([a_index, b_index])\n",
+    "                p_values.append(p)\n",
+    "                ratio_idxs.append(j)\n",
+    "                xaxis.append(names[categories[i]])\n",
+    "                i_list.append(i)\n",
+    "                \n",
+    "# Only retaining inforamtion for distances that are statistically significantly different, and are not people\n",
```

Comment:
this comment is wrong? it's retaining people, but not doing stat sig check here, right?